recipes/hdf5/meta.yaml

```diff
@@ -0,0 +1,36 @@
+package:
+  name: hdf5
+  version: 1.8.16
+
+source:
+  fn: hdf5-1.8.16.tar.gz
+  url: http://www.hdfgroup.org/ftp/HDF5/current/src/hdf5-1.8.16.tar.gz
+  md5: b8ed9a36ae142317f88b0c7ef4b9c618
+
+
+build:
+  features:
+    - vc9     # [win and py27]
+    - vc10    # [win and py34]
+    - vc14    # [win and py35]
```

Comment:
@jakirkham I'm excited to have so many brilliant people thinking about these issues.  I'll do what I can to help get @ukoethe's work with the looser constraints into conda-build.

Conda forge is a different story than conda-build, obviously.  For conda-forge, the question I think is "do you want to leverage conda-forge to support non-standard Python distributions?"  My sentiment is "probably," but at what cost in terms of taxing the build services?  If that cost is acceptable, this would be really exciting.  You could put together alternative non-standard distributions, like the one that @jjhelmus posted last week (ACPD?)